src/main/java/io/tiledb/java/api/Array.java

```diff
@@ -742,6 +743,26 @@ public BigInteger getMetadataNum() throws TileDBError {
     return new Pair<String, NativeArray>(keyString, result);
   }
 
+  /**
+   * Returns a HashMap with all array metadata in a key-value manner.
+   *
+   * @return The metadata
+   * @throws TileDBError
+   */
+  public Map<String, Object> getMetadataMap() throws TileDBError {
```

Comment:
Hmm, so we need to know the metadata type for the each key ahead of time?

Reply:
The idea behind that is that we can get the class of each object returned using `.getClass`. We could also provide the type as well, by returning something like `Pair<NativeArray, Datatype>` or `Pair<Object, Datatype>` but this would make the API more complex, without providing something more than `.getClass()` provides.

So we need either know the metadata type ahead of time or consider do some type-check before reading each value.

Reply:
OH, Java! Forgetting about it always; LGTM than! I will remove this method from my PR than and can leave only a simple putMetadata overload :D

Reply:
Sounds good! You can take a look here on how to use the `.getClass()` method: https://docs.oracle.com/javase/tutorial/reflect/class/classNew.html

You can also take a look on how `getClass()` is used in that unit test: https://github.com/TileDB-Inc/TileDB-Java/pull/157/files#diff-d38ae1adae50d20e357b8edd3c64de26R554

Reply:
@gsvic thanks ;) I'm just looking at everything through the prism of a typed API (got used to it after years in Scala).